retrieval_service/datastore/providers/neo4j_graph.py

```diff
@@ -110,7 +114,15 @@ async def search_airports(
         raise NotImplementedError("This client does not support airports.")
 
     async def get_amenity(self, id: int) -> Optional[models.Amenity]:
-        raise NotImplementedError("This client does not support amenities.")
+        async with self.__driver.session() as session:
+            result = await session.run("MATCH (a: Amenity {id: $id}) RETURN a", id=id)
+            record = await result.single()
+
+            if not record:
+                return None
+
+            amenity_data = dict(record["a"])
```

Comment:
Should we use record.data("a")? https://neo4j.com/docs/api/python-driver/current/api.html#neo4j.Record.data